bin/collator/src/parachain/chain_spec/mod.rs

```diff
@@ -0,0 +1,33 @@
+use sc_chain_spec::ChainSpecExtension;
+use serde::{Deserialize, Serialize};
+
+pub mod shibuya;
+pub mod shiden;
+
+pub use shibuya::ShibuyaChainSpec;
+pub use shiden::ShidenChainSpec;
+
+/// Node `ChainSpec` extensions.
+///
+/// Additional parameters for some Substrate core modules,
+/// customizable from the chain spec.
+#[derive(Default, Clone, Serialize, Deserialize, ChainSpecExtension)]
+#[serde(rename_all = "camelCase")]
+pub struct Extensions {
+    /// The relay chain of the Parachain.
+    pub relay_chain: String,
+    /// The id of the Parachain.
+    pub para_id: u32,
+}
+
+impl Extensions {
+    /// Try to get the extension from the given `ChainSpec`.
+    pub fn try_get(chain_spec: &dyn sc_service::ChainSpec) -> Option<&Self> {
+        sc_chain_spec::get_extension(chain_spec.extensions())
+    }
+}
+
+// TODO: add helper methods here?
+// The problem is that they don't have the same type so how to best handle it?
+// Generics seem a bit of an overkill since I need to introduce dependency on signature pallets then to implement the bound?
+// Introduction of a shared type? Could this also be useful for the SmartContract struct which is copy/pasted across different crates?
```

Comment:
I believe we don’t need to share this type between runtimes. Much more convenient to use it with runtime name prefix, like `shiden::Signature`

Reply:
Applying to AccountPublic IMO it’s not a big deal to keep it private type for each module, comparing to grown complexity because of increase count of references just to make it a bit more general.

Reply:
Ok, sure.

I don't have a strong opinion here but I was slightly leaning towards extracting types.
For the avoidance of code duplication that is.
Lower code complexity is a good argument though.